lib/msal-node/src/network/HttpClient.ts

```diff
@@ -105,6 +105,7 @@ const networkRequestViaProxy = <T>(
         request.on("connect", (response, socket) => {
             const statusCode = response?.statusCode || 500;
             if (statusCode < 200 || statusCode > 299) {
```

Comment:
Do we need to also remove this if-statement when connection to the proxy fails? @hectormmg

Reply:
Or maybe just resolve the error?

Reply:
My rationale behind keeping this as such was :  request.on("connect", (response, socket)) is essentially dealing with connection to the proxy alone, all other cases are handled down later, including an error stream on connection to the socket and timeout error. 
Testing with proxy to confirm. 
@Robbie-Microsoft @hectormmg

Reply:
Maybe we can change the error message to include: "Error connecting to proxy: [...]"